packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -206,6 +211,11 @@ class ForcePressGestureRecognizer extends OneSequenceGestureRecognizer {
 
   @override
   void addPointer(PointerEvent event) {
+    if (!isPointerAllowed(event)) {
```

Comment:
Seems like there ought to be a way to have the superclass enforce this.  Maybe have subclasses implement a protected `doAddPointer` and the superclass implementation does this check and then calls `doAddPointer`?